### PR TITLE
Fix SB and EH Message activity source names

### LINKED ISSUE #39166: [BUG] OpenTelemetry - ServiceBus message spans are reported in Azure.Messaging.ServiceBus activity source (same for EH)
### Library name and version

Azure.Messaging.ServiceBus, Azure.Messaging.EventHubs, Azure.Core

### Describe the bug

The convention we have is to use `<root namespace>.<client name>` for ActivitySource names so that users can enable individual libraries using `AddSource("<root namespace>.*")` and can even enable tracing on individual clients if necessary.

Message spans are created in a different way than service method spans, and due to this logic 

https://github.com/Azure/azure-sdk-for-net/blob/aa5fe51801498a3d967b85caaeb3bd53c4d3a2ad/sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs#L102-L107

end up with  `Azure.Messaging.ServiceBus|EventHubs` source name, not following the above convention (`<root namespace>.*`)

### Expected behavior

Message spans are special - they are coming from sender client, but we'd like to be able to enable them independently from sender spans, so ideally they should be reported by `Azure.Messaging.ServiceBus|EventHubs.Message` (or similar) sources.

### Actual behavior

ServiceBus and EventHubs report message spans from `Azure.Messaging.ServiceBus|EventHubs` sources.

### Reproduction Steps

1. Create simple ServiceBus producer app
2. Listen to all activity sources and print their names 
    ```csharp
    ActivitySource.AddActivityListener(new ActivityListener()     {
        ShouldListenTo = l => { Console.WriteLine(l.Name); return false; }});
    ```
3. Send some messages and check the sources:

```
Azure.Messaging.ServiceBus
Azure.Messaging.ServiceBus.ServiceBusSender
```

### Environment

_No response_

### PR DESCRIPTION
**Pull request by lmolkova**
Fix #39166

Change how `ActivitySource` name generated when activity name does not have a dot in it.
Prereq for https://github.com/Azure/azure-sdk-for-net/issues/39431

This is a breaking change. TODO:
- [x] check that AppInsights SDK is not broken
- [x] changelog

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs [hunk 1]
```diff
@@ -38,9 +35,10 @@ internal static class DiagnosticProperty
 
         /// <summary>
         ///   The activity name associated with events.
+        ///   TODO (limolkova): check it does not break appinsights SDK/visualizations
         /// </summary>
         ///
-        public static string EventActivityName { get; } = "EventHubs.Message";
```
Comment (lmolkova) [inline sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: This is part of #39166, but I don't really like this change and would rather rename SB one to include ServiceBus.

The problem is that this one now comes from we have `Azure.Messaging.EventHubs.EventHubs` source and if we rename SB one it'd come from `ServiceBus.Servicebus` 🤣

Comment (lmolkova) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: Changed to `EventHubMessage.Create`, checked AppInsights - it does not care, naming suggestions are very welcome

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: Why not use `Message` for both, if we're already fully qualifying with the root namespace?   `Azure.Messaging.EventHubs.Message` and `Azure.Messaging.ServiceBus.Message`?  

Or, should we consider using `event` to describe Event Hubs messages, which is what they are identified as in that ecosystem.

The `Create` part of the suggestion may be confusing, as we're not instrumenting the constructor.  Is that intended to mean "we sent to the service and the service created the data?"

File: sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             bool isActivitySourceEnabled = IsActivitySourceEnabled;
 
             scope.AddAttribute("http.method", message.Request.Method.Method);
-            scope.AddAttribute("http.url", _sanitizer.SanitizeUrl(message.Request.Uri.ToString()));
+            scope.AddAttribute("http.url", message.Request.Uri, u => _sanitizer.SanitizeUrl(u.ToString()));
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs:67]: Is this change related to the PR?

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: I thought if we didn't include the dot in the activity name, the whole activity name would be included in the activity source name per the changes in this PR. So wouldn't just using `Message` as the activity name (as Jesse suggested) give us the activity source we want?

Comment (lmolkova) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: > So wouldn't just using Message as the activity name (as Jesse suggested) give us the activity source we want?

Absolutely, but then it'd be inconsistent with every other span we create and the guidelines where we tell to do `Client.Method`.

For the consistency perspective, I'd prefer the current naming (with potential corrections around Message/Event for EH)

File: sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs (hunk 2 above)
Comment (lmolkova) [reply sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs:67]: nope, moving it away to #39637

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: But these spans are different as there isn't really a specific client or method to assign.

Comment (lmolkova) [reply sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/DiagnosticProperty.cs:43]: ok, will change back:) 
We'll also have displayName that follows otel semconv naming recommendations and at that point activity name will be deep technical detail.

**Review by JoshLove-msft — APPROVED** [review]

**Review by jsquire — APPROVED** [review]

**Review by annelo-msft — APPROVED** [review]
Would it be possible update the Azure.Core CHANGELOG as well?
